### PR TITLE
Remove os:exec dependency and introduce extern functions

### PR DESCRIPTION
**Pull request by lahirusamith**

### CODE REVIEW

File: sql-ballerina/Ballerina.toml [hunk 1]
```diff
@@ -28,3 +28,9 @@ groupId = "org.hsqldb"
 scope = "testOnly"
 artifactId = "hsqldb"
 version = "2.5.1"
+
+[[platform.java11.dependency]]
+path = "./lib/io-native-@io.native.version@.jar"
+groupId = "org.ballerinalang"
+artifactId = "io-native"
+version = "@io.native.version@"
```
Comment (daneshk) [inline sql-ballerina/Ballerina.toml:36]: why we need to add io native dependency in SQL module?

File: sql-ballerina/tests/process.bal [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return nativeExitCode(self);
+    }
+
+    # Provides a channel (to read from), which is made available as the 'standard error' of the process.
+    # ```ballerina
+    # io:ReadableByteChannel input = process.stderr();
+    # ```
+    #
+    # + return - The `io:ReadableByteChannel`, which represents the process's 'standard error'
+    public isolated function stderr() returns io:ReadableByteChannel {
```
Comment (daneshk) [inline sql-ballerina/tests/process.bal:54]: Seems like we are not using this function. shall we add only functions we need to exec docker commands

File: sql-ballerina/Ballerina.toml (hunk 1 above)
Comment (lahirusamith) [reply sql-ballerina/Ballerina.toml:36]: This dependency is needed to cater for the extern function stderr(). If stderr() is going to remove, then this dependency can be removed as well.

File: sql-ballerina/tests/process.bal (hunk 2 above)
Comment (lahirusamith) [reply sql-ballerina/tests/process.bal:54]: Okey, I will remove the stderr() function and related io-native dependency.

File: sql-ballerina/build.gradle [hunk 3]
```diff
@@ -100,6 +100,7 @@ task updateTomlFile {
 
         def newConfig = ballerinaConfigFile.text.replace("@project.version@", project.version.toString())
         newConfig = newConfig.replace("@toml.version@", tomlVersion)
+        newConfig = newConfig.replace("@io.native.version@", project.stdlibIoVersion)
```
Comment (daneshk) [inline sql-ballerina/build.gradle:103]: Shall we remove this

File: sql-ballerina/tests/utils.bal [hunk 4]
```diff
@@ -17,26 +17,28 @@
...
 string urlPrefix = "jdbc:hsqldb:hsql://localhost:";
 
-function initializeDockerContainer(string containerName, string dbAlias, string port, string resFolder, string scriptName) {
-    os:Process process;
+function initializeDockerContainer(string containerName, string dbAlias, string port, string resFolder,
+        string scriptName) {
     int exitCode = 1;
-    process = checkpanic os:exec(
-        "docker", {}, scriptPath, "run", "--rm", "-d", "--name", containerName,
+    setModuleForTest();
```
Comment (niveathika) [inline sql-ballerina/tests/utils.bal:32]: Wouldnt this be called for every container start?

File: sql-ballerina/build.gradle (hunk 3 above)
Comment (lahirusamith) [reply sql-ballerina/build.gradle]: sure!

File: sql-ballerina/tests/utils.bal [hunk 5]
```diff
@@ -17,26 +17,38 @@
 import ballerina/io;
 import ballerina/file;
 import ballerina/lang.runtime as runtime;
-import ballerina/os;
```
Comment (niveathika) [inline sql-ballerina/tests/utils.bal:20]: If this dependency is not used, lets remove it from the build

Comment (lahirusamith) [reply sql-ballerina/tests/utils.bal:20]: another module may be using it. can not remove at the moment.

**Review by niveathika — APPROVED** [review]

**Pull request merged**